provider/pkg/resources/custom.go

```diff
@@ -12,11 +13,11 @@ type CustomResource interface {
 	// Check validates and transforms the inputs of the resource.
 	Check(ctx context.Context, urn resource.URN, randomSeed []byte, inputs, state resource.PropertyMap, defaultTags map[string]string) (resource.PropertyMap, []ValidationFailure, error)
 	// Create creates a new resource in the cloud provider and returns its unique identifier and outputs.
-	Create(ctx context.Context, urn resource.URN, inputs resource.PropertyMap) (identifier *string, outputs map[string]any, err error)
+	Create(ctx context.Context, urn resource.URN, inputs resource.PropertyMap, timeout time.Duration) (identifier *string, outputs resource.PropertyMap, err error)
```

Comment:
This seems like a breaking change but we are expecting this interface to be internal? .. We're on `v1.5.0`. Nothing consumes it in other codebases? Should we work harder here as in introducing a side-by-side interface?

Reply:
Yeah, this interface is internal. There's no way somebody could use this interface and plug it into aws-native unless they fork aws-native. It's used internally to wrap arbitrary custom resources: https://github.com/pulumi/pulumi-aws-native/blob/0166a6357100a309d815a53cc4ac986c3fa96aab/provider/pkg/provider/provider.go#L509-L511